core/tauri/src/test/mod.rs

```diff
@@ -85,13 +142,120 @@ pub fn mock_context<A: Assets>(assets: A) -> crate::Context<A> {
   }
 }
 
-pub fn mock_app() -> crate::App<MockRuntime> {
-  crate::Builder::<MockRuntime>::new()
-    .build(mock_context(noop_assets()))
-    .unwrap()
+/// Creates a new [`Builder`] using the [`MockRuntime`].
+///
+/// To use a dummy [`Context`], see [`mock_app`].
+///
+/// # Examples
+///
+/// ```rust
+/// #[cfg(test)]
+/// fn do_something() {
+///   let app = tauri::test::mock_builder()
+///     // remove the string argument to use your app's config file
+///     .build(tauri::generate_context!("test/fixture/src-tauri/tauri.conf.json"))
+///     .unwrap();
+/// }
+/// ```
+pub fn mock_builder() -> Builder<MockRuntime> {
+  Builder::<MockRuntime>::new()
+}
+
+/// Creates a new [`App`] for testing using the [`mock_context`] with a [`noop_assets`].
+pub fn mock_app() -> App<MockRuntime> {
+  mock_builder().build(mock_context(noop_assets())).unwrap()
+}
+
+/// Executes the given IPC message and assert the response matches the expected value.
+///
+/// # Examples
+///
+/// ```rust
+/// #[tauri::command]
+/// fn ping() -> &'static str {
+///   "pong"
+/// }
+///
+/// fn create_app<R: tauri::Runtime>(mut builder: tauri::Builder<R>) -> tauri::App<R> {
+///   builder
+///     .invoke_handler(tauri::generate_handler![ping])
+///     // remove the string argument on your app
+///     .build(tauri::generate_context!("test/fixture/src-tauri/tauri.conf.json"))
+///     .expect("failed to build app")
+/// }
+///
+/// fn main() {
+///   let app = create_app(tauri::Builder::default());
+///   // app.run(|_handle, _event| {});}
+/// }
+///
+/// //#[cfg(test)]
+/// mod tests {
+///   use tauri::Manager;
+///
+///   //#[cfg(test)]
+///   fn something() {
+///     let app = super::create_app(tauri::test::mock_builder());
+///     let window = app.get_window("main").unwrap();
+///
+///     // run the `ping` command and assert it returns `pong`
+///     tauri::test::assert_ipc_response(
+///       &window,
+///       tauri::InvokePayload {
+///         cmd: "ping".into(),
+///         tauri_module: None,
+///         callback: tauri::api::ipc::CallbackFn(0),
+///         error: tauri::api::ipc::CallbackFn(1),
+///         inner: serde_json::Value::Null,
+///       },
+///       // the expected response is a success with the "pong" payload
+///       // we could also use Err("error message") here to ensure the command failed
+///       Ok("pong")
+///     );
+///   }
+/// }
+/// ```
+pub fn assert_ipc_response<T: Serialize>(
```

Comment:
This one also needs a timeout parameter.. and also it cannot be used with concurrency (only one assert call at a time because we check the last evaluated JS script). Maybe we should use a builder instead (ugly).